src/development/dev-tests-coding.md

```diff
@@ -0,0 +1,165 @@
+  As you read the test cases, you will need to go back to the shared data and keep in your head what user1 and user2 are being used for. How they are used is down to the test case. The lack of intent from the field definitions means the reader has to keep this context in their head.
+
+A better approach is to declare the fixture within the test.
+
+## Create fixture data in factory methods
+
+If we’ve decided to avoid shared fixture data, then won’t we end up with lots of object creation code within our tests? Yes, you can. This is ok if object creation can be done with clear intent in a one liner. But even using lombok builders, object creation can often span multiple lines which leads to needless boilerplate.
+
+A very simple solution to this is to use simple factory methods at the bottom of the test class.
+
+* __Less verbose__
+  Exposing a method that only takes the argument that varies for the test case means all the default boilerplate is handled in one place, out of the way.
+* __Clear intent__
+  Exposing multiple methods to describe different situations can clearly express why a fixture object is being created.
+* __Clear naming__
+  As objects are created within the test, they can be assigned to variables that suit the test case.
+
+Use methods with clear names and intent to create fixture data when it is needed.
+Name the created fixture objects to further clarify the intent within the test.
+
+## Testing with date and time
+
+When code is dependent on date or time, your tests must take control of time so they can test all the different flows and expectations.
+
+We don't hardcode fixture data to specific dates/times far in the future. This test will fail one day and another poor developer will have to come along and mop up the problem.
+
+There are two common approaches to testing with time:
+
+* __Offset input data__
+  Use system date and/or time and adjust with an offset to create values in fixture data. The code will apply logic based on the current system time and the input data. This is a simple approach that works well in most cases.
+
+* __Control system time__
+  Sometimes you can't use fixture data: For example, in the [Greeter kata](https://github.com/wix/tdd-katas), a salutation changes through the course of the day. We obviously can't write tests that only pass in the morning. Instead we need to control system time.
+  
+  Most languages have a way to control system time in code.
+  * [Java](https://www.baeldung.com/java-override-system-time)
+  * [Javascript using Jest](https://codewithhugo.com/mocking-the-current-date-in-jest-tests/)
+
+## Test one thing
+
+If you adopt Test Driven Development (TDD) you should find that your test cases are lean, with just enough fixture data to test a single additional requirement.
+
+If you find that each additional test case is actually performing all the previous steps to initialise your fixtures, then you're probably doing too much work, and diluting the intent of the tests.
```

Comment:
Is there any guidance on how long a particular unit test should run?  Ideally I'd want to be running the tests everytime I save the file but if the unit tests are taking an age then it might be a sign they're doing too much.